### PR TITLE
feat: Implementing the possibility to remove a friend

### PR DESCRIPTION
**Pull request by yassine04e**
1) Adding a list of friends in the modifiying profile screen, with clickable cards to delete a friend 
2) Adding the function in the repository. It uses transactions

### CODE REVIEW

File: app/src/main/java/com/github/se/travelpouch/ui/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -103,54 +102,55 @@ fun ProfileScreen(navigationActions: NavigationActions, profileModelView: Profil
 
           LazyColumn(
               modifier = Modifier.fillMaxSize().padding(horizontal = 8.dp),
-              contentPadding = PaddingValues(bottom = 80.dp)) {
-                val friends = profile.value.friends.keys.toList()
+              // contentPadding = PaddingValues(bottom = 80.dp)
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/ui/profile/ProfileScreen.kt:105]: Make sure not to leave any commented out lines like this

File: app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepositoryFirebase.kt [hunk 2]
```diff
@@ -217,6 +217,46 @@ class ProfileRepositoryFirebase(private val db: FirebaseFirestore) : ProfileRepo
...
+                  t.update(documentReference!!, "friends", userProfileUpdated.friends)
+                }
+                .addOnSuccessListener {
+                  Log.d("DeleteFriend", "Friend deleted")
+                  onSuccess(userProfileUpdated)
+                }
+                .addOnFailureListener {
+                  Log.e("DeleteFriend", "An error occurred updating user profile")
+                  onFailure(Exception("An error occurred updating your profile"))
+                }
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepositoryFirebase.kt:251]: Good use of transactions, and the error handling is succinct and logical.

File: app/src/main/java/com/github/se/travelpouch/ui/profile/ModifyingProfileScreen.kt [hunk 3]
```diff
@@ -140,74 +148,163 @@ fun ModifyingProfileScreen(
...
+                            }
+                      }
+                }
+              }
+
+              if (openDialogToDeleteFriend) {
+                Dialog(onDismissRequest = { openDialogToDeleteFriend = false }) {
+                  Box(
+                      Modifier.fillMaxWidth(1f)
+                          .height(250.dp)
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/ui/profile/ModifyingProfileScreen.kt:272]: While it's the same size as for when adding a friend, the dialog box is too big for the content inside.
Either increase the size of the content or reduce the size of the dialog box. It feels weird.
Screenshot_2024-12-05_20-52-01

File: app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepositoryFirebase.kt [hunk 4]
```diff
@@ -217,6 +217,46 @@ class ProfileRepositoryFirebase(private val db: FirebaseFirestore) : ProfileRepo
         .addOnFailureListener { onFailure(Exception("getting friend profile failed")) }
   }
 
+  override fun removeFriend(
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepositoryFirebase.kt:220]: Please add some minimal documentation for this function

File: app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepository.kt [hunk 5]
```diff
@@ -55,4 +55,11 @@ interface ProfileRepository {
       onSuccess: (Profile) -> Unit,
       onFailure: (Exception) -> Unit
   )
+
+  fun removeFriend(
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepository.kt:59]: Please add minimal documentation to this function

File: app/src/main/java/com/github/se/travelpouch/model/profile/ProfileModelView.kt [hunk 6]
```diff
@@ -85,4 +85,15 @@ class ProfileModelView @Inject constructor(private val repository: ProfileReposi
           onFailure(it)
         })
   }
+
+  fun removeFriend(friendUid: String, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/profile/ProfileModelView.kt:89]: Please add minimal documentation to this function

**Review by RemIsMyWaifuu — CHANGES_REQUESTED** [review]
Apart from some UI issues that need to be resolved, the backend and tests are sublime.

**Comment by RemIsMyWaifuu** [comment]
Furthermore, the text in ProfileScreen is not aligned the same way as in ModifyingProfileScreen. This causes the "Friends :" text to be aligned on the left, but then aligned in the middle once a user enters editing mode. This is strange.

`Column(
        verticalArrangement = Arrangement.spacedBy(8.dp),
        modifier =
            Modifier.fillMaxWidth()
                .padding(horizontal = 16.dp, vertical = 16.dp)
                .padding(pd)
                .testTag("ProfileColumn"))`
and

`Column(
            verticalArrangement = Arrangement.spacedBy(8.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier =
                Modifier.fillMaxWidth()
                    .padding(horizontal = 16.dp, vertical = 16.dp)
                    .padding(pd)
                    .testTag("ProfileColumn")
                    .verticalScroll(rememberScrollState()))`

**Comment by RemIsMyWaifuu** [comment]
Also, the add friend button is on the bottom right. On the edit travel screen the way we decided to address this was to make a drawer button. Since this is not necessary here, please change the colour to be the same blend as in EditTravelSettings and [Optional] perhaps put the button under the friends list instead.

File: app/src/main/java/com/github/se/travelpouch/ui/profile/ProfileScreen.kt (hunk 1 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/ui/profile/ProfileScreen.kt:105]: This is done

File: app/src/main/java/com/github/se/travelpouch/ui/profile/ModifyingProfileScreen.kt (hunk 3 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/ui/profile/ModifyingProfileScreen.kt:272]: I have corrected it

File: app/src/main/java/com/github/se/travelpouch/model/profile/ProfileModelView.kt (hunk 6 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/profile/ProfileModelView.kt:89]: Done

File: app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepository.kt (hunk 5 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepository.kt:59]: Done

File: app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepositoryFirebase.kt (hunk 4 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/profile/ProfileRepositoryFirebase.kt:220]: Done

**Comment by yassine04e** [comment]
Concerning the two other comments, I have applied the modifications you requested. I changed the floating button to add a friend, to instead put a button similar to the one to save our profile. It is placed just above the save button. 
Also, I have corrected the horizontal alignement of the column in the profile screen. Now it is the same as the edit profile screen.

**Review by RemIsMyWaifuu — APPROVED** [review]

**Comment by RemIsMyWaifuu** [comment]
While technically the scope of this PR concerns itself only with friend removal, when a user saves their profile all their friends disappear..
All the other requested changes were addressed, I also removed some commented lines.
Approved.
Closes #238

**Pull request merged**